src/libpanic_unwind/emcc.rs

```diff
@@ -52,22 +52,32 @@ pub fn payload() -> *mut u8 {
     ptr::null_mut()
 }
 
+struct Exception {
+    data: Option<Box<dyn Any + Send>>,
```

Comment:
Reading over this it seemed odd that the `Option` went away below but seemed to be added here, but this is for `emcc.rs` so I'm not reviewing it too too closely.

Reply:
We added an option here because the C++ runtime will invoke the destructor during `__cxa_end_catch`. We need to ensure that the exception object contains `None` at that point to avoid a double-free.